### ISSUE TITLE
Unhandled rejection TypeError: (0 , _kuromojin2.default) is not a function

### ISSUE THREAD

**Issue by Jxck** [textlint/textlint#196]
## command

```sh
$ DEBUG=textlint* ./node_modules/.bin/textlint -c .textlintrc readme.md
Wed, 27 Apr 2016 16:09:20 GMT textlint:cli Running on files, stdin-filename: undefined
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader config %O Config {
  configFile: '/Users/jxck/dotfiles/misc/.textlintrc',
  rulesBaseDirectory: undefined,
  rules: 
   [ 'spellcheck-tech-word',
     'no-mix-dearu-desumasu',
     'no-double-negative-ja',
     'no-dropping-the-ra',
     'no-doubled-conjunctive-particle-ga',
     'incremental-headers',
     'no-doubled-joshi',
     'sentence-length',
     'no-start-duplicated-conjunction',
     'max-ten',
     'max-length-of-title',
     'max-appearence-count-of-words' ],
  disabledRules: [],
  presets: [],
  plugins: [],
  rulesConfig: 
   { 'spellcheck-tech-word': true,
     'no-mix-dearu-desumasu': true,
     'no-double-negative-ja': true,
     'no-dropping-the-ra': true,
     'no-doubled-conjunctive-particle-ga': true,
     'incremental-headers': true,
     'no-doubled-joshi': { min_interval: 1, strict: false },
     'sentence-length': { max: 100 },
     'no-start-duplicated-conjunction': { interval: 2 },
     'max-ten': { max: 5 },
     'max-length-of-title': { '#': 40, '##': 30 },
     'max-appearence-count-of-words': { limit: 5 } },
  extensions: [],
  rulePaths: [],
  formatterName: undefined,
  color: true }
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-spellcheck-tech-word/spellcheck-tech-word.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-mix-dearu-desumasu/lib/no-mix-dearu-desumasu.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-double-negative-ja/lib/no-doubled-negative-ja.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-dropping-the-ra/lib/no-dropping-the-ra.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-doubled-conjunctive-particle-ga/lib/no-doubled-conjunctive-particle-ga.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-incremental-headers/lib/incremental-headers.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-doubled-joshi/lib/no-doubled-joshi.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-sentence-length/lib/sentence-length.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-start-duplicated-conjunction/lib/no-start-duplicated-conjunction.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-max-ten/lib/max-ten.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-max-length-of-title/lib/max-length-of-title.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-max-appearence-count-of-words/lib/max-appearence-count-of-words.js
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "spellcheck-tech-word" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "no-mix-dearu-desumasu" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "no-double-negative-ja" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "no-dropping-the-ra" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "no-doubled-conjunctive-particle-ga" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "incremental-headers" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "no-doubled-joshi" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "sentence-length" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "no-start-duplicated-conjunction" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "max-ten" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "max-length-of-title" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:rule-creator-set use "max-appearence-count-of-words" rule
Wed, 27 Apr 2016 16:09:20 GMT textlint:find-util Processing blog.jxck.io/entries/2016-04-16/stale-while-revalidate.md
Wed, 27 Apr 2016 16:09:20 GMT textlint:core-task max-length-of-title pushReport {}
Unhandled rejection TypeError: (0 , _kuromojin2.default) is not a function
at /Users/jxck/jxck.io/blog.jxck.io/entries/2016-04-16/stale-while-revalidate.md
    at /Users/jxck/jxck.io/node_modules/textlint-rule-no-dropping-the-ra/lib/no-dropping-the-ra.js:20:40
    at /Users/jxck/jxck.io/node_modules/carrack/lib/index.js:72:32
    at Array.forEach (native)
    at RuleTypeEmitter.emit (/Users/jxck/jxck.io/node_modules/carrack/lib/index.js:71:29)
    at Controller.enter (/Users/jxck/jxck.io/node_modules/textlint/lib/task/textlint-core-task.js:159:55)
    at Controller.Execute [as __execute] (/Users/jxck/jxck.io/node_modules/txt-ast-traverse/lib/txt-ast-traverse.js:54:29)
    at Controller.traverse (/Users/jxck/jxck.io/node_modules/txt-ast-traverse/lib/txt-ast-traverse.js:129:24)
    at TextLintCoreTask.start (/Users/jxck/jxck.io/node_modules/textlint/lib/task/textlint-core-task.js:154:32)
    at /Users/jxck/jxck.io/node_modules/textlint/lib/task/task-runner.js:44:22
    at Function.process (/Users/jxck/jxck.io/node_modules/textlint/lib/task/task-runner.js:32:20)
    at LinterProcessor.process (/Users/jxck/jxck.io/node_modules/textlint/lib/linter/linter-processor.js:67:41)
    at TextlintCore._parallelProcess (/Users/jxck/jxck.io/node_modules/textlint/lib/textlint-core.js:170:36)
    at /Users/jxck/jxck.io/node_modules/textlint/lib/textlint-core.js:226:31
    at tryCatcher (/Users/jxck/jxck.io/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/jxck/jxck.io/node_modules/bluebird/js/release/promise.js:502:31)
    at Promise._settlePromise (/Users/jxck/jxck.io/node_modules/bluebird/js/release/promise.js:559:18)
```

## package.json

dependencies only

```json
{
  "rules": {
    "spellcheck-tech-word": true,
    "no-mix-dearu-desumasu": true,
    "no-double-negative-ja": true,
    "no-dropping-the-ra": true,
    "no-doubled-conjunctive-particle-ga": true,
    "incremental-headers": true,
    "no-doubled-joshi": {
      "min_interval" : 1,
      "strict": false
    },
    "sentence-length": {
      "max": 100
    },
    "no-start-duplicated-conjunction": {
      "interval": 2
    },
    "max-ten": {
      "max": 5
    },
    "max-length-of-title": {
      "#": 40,
      "##": 30
    },
    "max-appearence-count-of-words": {
      "limit": 5
    }
  }
}
```

## textlintrc

```json
{
  "dependencies": {
    "remark": "^4.2.2"
  },
  "devDependencies": {
    "gulp": "^3.9.1",
    "gulp-image": "^1.3.1",
    "textlint": "^6.3.0",
    "textlint-rule-incremental-headers": "^0.2.0",
    "textlint-rule-max-appearence-count-of-words": "^1.0.1",
    "textlint-rule-max-length-of-title": "^1.0.1",
    "textlint-rule-max-ten": "^2.0.1",
    "textlint-rule-no-double-negative-ja": "^1.0.3",
    "textlint-rule-no-doubled-conjunctive-particle-ga": "^1.0.2",
    "textlint-rule-no-doubled-joshi": "^3.2.0",
    "textlint-rule-no-dropping-the-ra": "^1.0.1",
    "textlint-rule-no-mix-dearu-desumasu": "^1.4.0",
    "textlint-rule-no-start-duplicated-conjunction": "^1.0.7",
    "textlint-rule-preset-japanese": "^1.0.3",
    "textlint-rule-preset-jtf-style": "^2.1.2",
    "textlint-rule-sentence-length": "^1.0.4",
    "textlint-rule-spellcheck-tech-word": "^5.0.0",
    "textlint-rule-web-plus-db": "^1.1.5"
  }
}
```

**Comment by azu** [comment]
@Jxck Thanks for report.
It is a bug of combination of kuromojin@1.2.0 and [textlint-rule-no-dropping-the-ra](https://github.com/azu/textlint-rule-no-dropping-the-ra "textlint-rule-no-dropping-the-ra").

I fix this and publish [textlint-rule-no-dropping-the-ra@1.0.2](https://github.com/azu/textlint-rule-no-dropping-the-ra/releases/tag/1.0.2) and [kuromojin@1.2.1](https://github.com/azu/kuromojin/releases/tag/1.2.1 "kuromojin")

Please try to update `textlint-rule-no-dropping-the-ra@1.0.2`

```
npm install -D textlint-rule-no-dropping-the-ra@1.0.2
```

**Comment by Jxck** [comment]
thanks for quick update.
but new error comes here.

```
Wed, 27 Apr 2016 17:01:48 GMT textlint:cli Running on files, stdin-filename: undefined
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader config %O Config {
  configFile: '/Users/jxck/dotfiles/misc/.textlintrc',
  rulesBaseDirectory: undefined,
  rules: 
   [ 'spellcheck-tech-word',
     'no-mix-dearu-desumasu',
     'no-double-negative-ja',
     'no-dropping-the-ra',
     'no-doubled-conjunctive-particle-ga',
     'incremental-headers',
     'no-doubled-joshi',
     'sentence-length',
     'no-start-duplicated-conjunction',
     'max-ten',
     'max-length-of-title',
     'max-appearence-count-of-words' ],
  disabledRules: [],
  presets: [],
  plugins: [],
  rulesConfig: 
   { 'spellcheck-tech-word': true,
     'no-mix-dearu-desumasu': true,
     'no-double-negative-ja': true,
     'no-dropping-the-ra': true,
     'no-doubled-conjunctive-particle-ga': true,
     'incremental-headers': true,
     'no-doubled-joshi': { min_interval: 1, strict: false },
     'sentence-length': { max: 100 },
     'no-start-duplicated-conjunction': { interval: 2 },
     'max-ten': { max: 5 },
     'max-length-of-title': { '#': 40, '##': 30 },
     'max-appearence-count-of-words': { limit: 5 } },
  extensions: [],
  rulePaths: [],
  formatterName: undefined,
  color: true }
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-spellcheck-tech-word/spellcheck-tech-word.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-mix-dearu-desumasu/lib/no-mix-dearu-desumasu.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-double-negative-ja/lib/no-doubled-negative-ja.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-dropping-the-ra/lib/no-dropping-the-ra.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-doubled-conjunctive-particle-ga/lib/no-doubled-conjunctive-particle-ga.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-incremental-headers/lib/incremental-headers.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-doubled-joshi/lib/no-doubled-joshi.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-sentence-length/lib/sentence-length.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-no-start-duplicated-conjunction/lib/no-start-duplicated-conjunction.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-max-ten/lib/max-ten.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-max-length-of-title/lib/max-length-of-title.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:module-loader Loading rules from /Users/jxck/jxck.io/node_modules/textlint-rule-max-appearence-count-of-words/lib/max-appearence-count-of-words.js
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "spellcheck-tech-word" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "no-mix-dearu-desumasu" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "no-double-negative-ja" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "no-dropping-the-ra" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "no-doubled-conjunctive-particle-ga" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "incremental-headers" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "no-doubled-joshi" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "sentence-length" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "no-start-duplicated-conjunction" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "max-ten" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "max-length-of-title" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:rule-creator-set use "max-appearence-count-of-words" rule
Wed, 27 Apr 2016 17:01:48 GMT textlint:find-util Processing blog.jxck.io/entries/2016-04-16/stale-while-revalidate.md
Wed, 27 Apr 2016 17:01:49 GMT textlint:core-task max-length-of-title pushReport {}
Wed, 27 Apr 2016 17:01:49 GMT textlint:core-task sentence-length pushReport {"line":0,"column":1}
Wed, 27 Apr 2016 17:01:49 GMT textlint:core-task sentence-length pushReport {"line":0,"column":1}
Unhandled rejection Error: unknown compression method
at /Users/jxck/jxck.io/blog.jxck.io/entries/2016-04-16/stale-while-revalidate.md
    at Zlib._handle.onerror (zlib.js:363:17)
```

**Comment by azu** [comment]
It seem to be issue of Node.js v6.0.0 and [kuromoji.js](https://github.com/takuyaa/kuromoji.js "kuromoji.js").
I'll see it lator.

**Comment by azu** [comment]
fmm, related issue? https://github.com/nodejs/node/issues/5852

**Comment by Jxck** [comment]
using v5.9.0 now.
I'll try with v6.0.0.

**Comment by Jxck** [comment]
v6.0.0 works fine for me.

**Comment by azu** [comment]
> Unhandled rejection Error: unknown compression method

This is a bug of Node.js v5.9.x. It is fixed 5.10 and 6.0.
I'll close this issue.

Related: 

- Issue: https://github.com/nodejs/node/issues/5852
- fixed: https://github.com/nodejs/node/pull/5863
- similar issue https://www.jfrog.com/jira/browse/RTFACT-9969
